HIP/hip-551.md

```diff
@@ -0,0 +1,89 @@
+This makes it impossible to create more complicated flows without using smart contracts (which do not support all the HAPI transactions at this point) and listening to the mirror node to check the status of the previous transaction.
+This way we can also achieve an abstraction away from smart contracts
+
+## Rationale
+Atomic transactions chain is from the protobuf point of view just a list of `SignedTransactions` that's, why we based our new protobuf message on it,
+we only have to change the `bodyBytes` field to be `repeated`, that way we can store there multiple transaction bodies,
+`SignatureMap sigMap` can stay unchanged because it's defined as `repeated SignaturePair sigPair`.
+
+## User stories
+
+As a Hedera token service user, I want to be able to unfreeze an account, send an NFT, and freeze it again in one ACID transaction, that way I can achieve an account-bound NFT(nontransferable NFT) collection, without using the hedera smart contract service.
+We can use nontransferable NFTs e.g. as someone’s achievements, the creation of digital references, etc.
+
+As a Hedera token service user, I want to be able to unfreeze an account, send an FT and freeze it again in one ACID transaction, that way I can achieve an account-bound FT(nontransferable FT) collection, without using the hedera smart contract service.
+We can use nontransferable FT e.g. as game points, reputation points, etc.
+
+As a Hedera token service user, I want to be able to wipe a token, mint a token, and transfer a token in one ACID transaction.
+That way a backend developer could create complicated flows without the need to use smart contracts and the need to handle partially successful flows in the backend,
+e.g. the wipe transaction was successful but the mint transaction was not.
+
+As a Hedera service user, I want to be able to create batch transaction, that use multiple Hedera services e. g.
+I want to be able to wipe a token, mint a token, transfer a token, and submit a topic message in one ACID transaction.
+That way a backend developer could create complicated flows without the need to use smart contracts and the need to handle partially successful flows in the backend,
+e.g. the wipe transaction was successful but the mint transaction was not.
+
+## Specification
+We could create a new protobuf message based on `SignedTransaction` message:
+```
+message BatchTransaction {
+    /**
+    * TransactionBodies serialized into bytes, which must be signed
+    */
+    repeated bytes bodyBytes = 1;
+
+    /**
+     * The signatures on the body with the new format, to authorize the transactions
+     */
+    SignatureMap sigMap = 2;
+}
+```
+We could also create a predefined set of batch transactions and add new elements to it in the future
+if the community asks for it.<br>
+Initial set:
+```
+Unfreeze -> Transfer -> Freeze
```

Comment:
Are there any non HTS cases ? Is this for HTS only or all HAPI transactions?
If there's a set on non applicable transactions those should be called out

Reply:
It would be more practical if these were for all HAPI transactions. Any application that needs a hapi transaction to succeed in order to do something else would use this. Gated topics or consensus messages have huge potential for content creation use cases. Here are a couple of scenarios:
Transfer -> Topic Create -> Submit Topic Message
Transfer -> Submit Topic Message

Reply:
The ability to combine HTS with HCS would be super powerful if they are batched. So, would also agree for all HAPI transactions to reach the maximum value out of this HIP proposal.